src/Codeception/Module/Doctrine2.php

```diff
@@ -424,7 +424,7 @@ protected function buildAssociationQuery($qb, $assoc, $alias, $params)
             if ($val === null) {
                 $qb->andWhere("s.$key IS NULL");
             } else {
-                $paramname = "s__$key";
+                $paramname = str_replace(".", "", "s_$key");
```

Comment:
When you are working with Embeddable object in Doctrine you must use notation with dot, for example:
$I->seeInRepository("Post", [
    'author.email' => "email@example.com",
]);

A variable $paramname is build based on $key so in above example will be: "s__author.email". A dot in this variable is a reason for exception:
[Doctrine\ORM\Query\QueryException] [Syntax Error] line 0, col 93: Error: Expected end of string, got '.' 

With this replace everything works great, I haven`t noticed any problems. I see all checks have passed so I hope you will merge this pull request.